packages/gatsby/src/services/initialize.ts

```diff
@@ -214,17 +207,15 @@ export async function initialize({
     )
   }
 
-  if (process.env.GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND) {
-    if (process.env.gatsby_executing_command !== `develop`) {
-      // we don't want to ever have this flag enabled for anything than develop
-      // in case someone have this env var globally set
-      delete process.env.GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND
-    } else if (isCI() && !process.env.CYPRESS_SUPPORT) {
-      delete process.env.GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND
-      reporter.verbose(
-        `Experimental Query on Demand feature is not available in CI environment. Continuing with eager query running.`
-      )
-    }
+  // Throughout the codebase GATSBY_QUERY_ON_DEMAND is used to conditionally enable the QoD behavior, depending on if "gatsby develop" is running or not. In CI QoD is disabled by default, too.
+  // You can use GATSBY_ENABLE_QOD_IN_CI to force enable it in CI.
+  process.env.GATSBY_QUERY_ON_DEMAND = `true`
+  if (process.env.gatsby_executing_command !== `develop`) {
+    // we don't want to ever have this flag enabled for anything than develop
+    // in case someone have this env var globally set
+    delete process.env.GATSBY_QUERY_ON_DEMAND
+  } else if (isCI() && !process.env.GATSBY_ENABLE_QOD_IN_CI) {
+    delete process.env.GATSBY_QUERY_ON_DEMAND
```

Comment:
Worth keeping the log? Something like (formatting may need adjustment):

```suggestion
    delete process.env.GATSBY_QUERY_ON_DEMAND
    reporter.verbose(
      `Query on Demand feature is not available in CI environment by default, continuing with eager query running. You can enable it by setting the GATSBY_ENABLE_QOD_IN_CI env var to a truthy value.`
    )
```